util/sha256/sha256.go

```diff
@@ -0,0 +1,144 @@
+// Package sha256 is like crypto/sha256,
```

Comment:
can we give this a different package name that doesn't collide with crypto/sha256?

Reply:
I called it `sha256fixed`. Happy to hear alternative names.